Follower terminology in the docs (#5067)
digital-asset/daml: docs/source/getting-started/first-feature.rst

Your First Feature
******************

> Let's dive into implementing a new feature for our social network app. 

Reviewer: It might be good to configure your editor to remove trailing spaces. Are you using VSCode?
```suggestion
Let's dive into implementing a new feature for our social network app.
```

The workflow aspect refers to the interactions between parties that are permitted by the system.
In the context of a messaging feature, these are essentially the authorization and privacy concerns listed above.

> For the authorization part, we take the following approach: a user Bob can message another user Alice when Alice starts following Bob back.

Reviewer: ```suggestion
For the authorization part, we take the following approach: a user Bob can message another user Alice only when Alice follows Bob.
```

> When Alice starts following Bob back, she gives permission or *authority* to Bob to send her a message.

Reviewer: ```suggestion
When Alice follows Bob, she gives permission or *authority* to Bob to send her a message.
```

To implement this workflow, let's start by adding the new *data* for messages.

> Navigate to the ``daml/User.daml`` file and copy the following ``Message`` template to the bottom. 

Reviewer: ```suggestion
Navigate to the ``daml/User.daml`` file and copy the following ``Message`` template to the bottom.
```

Messaging UI
============

The UI for messaging will consist of a new *Messages* panel in addition to the *Follow* and *Network* panel.
This panel will have two parts:

    1. A list of messages you've received with their senders.

>     2. A form with a dropdown menu for user selection and a text field for composing the message.

Reviewer: ```suggestion
    2. A form with a dropdown menu for follower selection and a text field for composing the message.
```

MessageEdit Component
---------------------

> Next we need the ``MessageEdit`` component to compose and send messages to selected users that we are following (and who are also following us back).

Reviewer: We don't need to follow somebody to message them. They give us the authority to message them by following us.
```suggestion
Next we need the ``MessageEdit`` component to compose and send messages to our followers.
```

The result of a successful call to ``submitMessage`` is a new ``Message`` contract created on the ledger.

The return value of this component is the React ``Form`` element.
This contains a dropdown menu to select a receiver from the ``following``, a text field for the message content, and a *Send* button which triggers ``submitMessage``.

There is again an important point here, in this case about how *authorization* is enforced.

> Due to the logic of the ``SendMessage`` choice, it is impossible to send a message to a user who has not followed us back (even if you could somehow access their ``User`` contract).

Reviewer: ```suggestion
Due to the logic of the ``SendMessage`` choice, it is impossible to send a message to a user who is not following us (even if you could somehow access their ``User`` contract).
```

First, navigate to the terminal window where the ``daml-start.sh`` process is running and terminate the active process by hitting ``Ctrl-C``. This shuts down the previous instances of the sandbox. Next in the root ``create-daml-app`` folder run ``./daml-start.sh``.

> As mentioned at the beginning of this *Getting Started with DAML* guide, DAML Sandbox uses an in-memory store, which means it loses its state when stopped or restarted. That means that all user data and list of ``following`` connections between them are lost. 

Reviewer: ```suggestion
As mentioned at the beginning of this *Getting Started with DAML* guide, DAML Sandbox uses an in-memory store, which means it loses its state when stopped or restarted. That means that all user data and follower relationships are lost. 
```

If you have the frontend UI up and running you're all set. In case you don't have the UI running open a new terminal window and navigate to the ``create-daml-app/ui`` folder and run the ``yaml start`` command, which will start the UI. 

Once you've done all these changes you should see the same login page as before at http://localhost:3000.
Once you've logged in, you'll see a familiar UI but with our new *Messages* panel at the bottom!
Go ahead and add follow more users, and log in as some of those users in separate browser windows to follow yourself back.

> Then, if you click on the dropdown menu in the *Messages* panel, you'll be able to see some users to message!

Reviewer: ```suggestion
Then, if you click on the dropdown menu in the *Messages* panel, you'll be able to see some followers to message!
```
